### PR TITLE
test: Replace TEST_PREVIOUS_RELEASES env var with test_framework option

### PR DESCRIPTION
**Pull request by maflcko**
The "auto-detection" feature is kept in place, but making it an option allows to properly document it. For example, on my machine I get:

```
$ ./test/functional/wallet_disable.py --help | grep previous-releases
  --previous-releases   Force test of previous releases (default: False)

### CODE REVIEW

**Comment by fanquake** [comment]
Concept ACK

**Comment by Sjors** [comment]
Concept ACK. You have to explicitly opt out of  `--previous-releases` on Travis, because `PREVIOUS_RELEASES_DIR` is created and cached for all machines.

File: test/functional/test_framework/test_framework.py [hunk 1]
```diff
@@ -154,6 +155,9 @@ def parse_args(self):
                             help="Print out all RPC calls as they are made")
         parser.add_argument("--portseed", dest="port_seed", default=os.getpid(), type=int,
                             help="The seed to use for assigning port numbers (default: current process id)")
+        parser.add_argument("--previous-releases", dest="prev_releases", action="store_true",
+                            default=os.path.isdir(previous_releases_path) and bool(os.path.listdir(previous_releases_path)),
```
Comment (Sjors) [inline test/functional/test_framework/test_framework.py:159]: That's a Python 3.6 syntax  I think
https://docs.python.org/3/library/os.html#os.listdir

Comment (MarcoFalke) [reply test/functional/test_framework/test_framework.py]: Nah, just me being stupid. I forgot to test this locally before pushing

**Comment by Sjors** [comment]
tACK fab56c1a6e95b75f0dacb2d7301ac0fec9a310d5

It would be nice if you could still opt-out without having to move the releases folder. This can be useful when a new version is added and I can't be bothered to download it.

**Comment by MarcoFalke** [comment]
> It would be nice if you could still opt-out without having to move the releases folder. This can be useful when a new version is added and I can't be bothered to download it.

If this auto-detection behavior is too brittle, we should simply remove it. It seems odd to have default values depend on the presence of directories or files on the hard disk.

**Comment by Sjors** [comment]
> If this auto-detection behavior is too brittle, we should simply remove it. It seems odd to have default values depend on the presence of directories or files on the hard disk.

That's annoying. Now you have to add `--previous-releases` every time to run those tests. Checking the presence of `/releases` for the default was better. We can always improve the error messages a bit, e.g. show which specific version is missing (you can already tell from the exceptions).

**Comment by Sjors** [comment]
re-tACK fab56c1 thanks for restoring

**Comment by MarcoFalke** [comment]
Rebased

**Comment by Sjors** [comment]
re-tACK fad798b

**Pull request merged**